lib/cn/kvset.c

```diff
@@ -1914,35 +1914,37 @@ kvset_get_vbsetv(struct kvset *ks, uint *vbsetc)
 }
 
 /**
- * kvset_get_max_key() - Get the largest key in a kvset
+ * kvset_get_min_key() - Get the smallest key in a kvset
  * @ks:   struct kvset handle.
- * @key:  (output) largest key. Null if kvset contains only ptombs.
- * @klen: (output) length of @key. Zero if kvset contains only ptombs.
+ * @min_key:  (output) smallest key
+ * @min_klen: (output) length of @min_key
  */
 void
-kvset_get_max_key(struct kvset *ks, void **key, uint *klen)
+kvset_get_min_key(struct kvset *ks, void **min_key, uint *min_klen)
 {
-    struct kvset_kblk *kb;
-    int                last = ks->ks_st.kst_kblks - 1;
+    struct kvset_kblk *kb = &ks->ks_kblks[0];
 
-    *key = 0;
-    *klen = 0;
+    INVARIANT(min_key && min_klen);
 
-    /* Check if kblock has non-ptombs. If not, use the second to last kblk.
-     * If there is exactly one kblock and it contains only ptombs, return
-     * with (*key) set to 0;
-     */
-    if (ks->ks_kblks[last].kb_wbt_desc.wbd_n_pages == 0) {
-        if (!last)
-            return;
+    *min_key = (void *)kb->kb_koff_min;
+    *min_klen = kb->kb_klen_min;
+}
```

Comment:
Is that cast needed?  Just curious...  Ah, it's because you're de-constifying kb_koff_min.  Seems ill-advised...

Reply:
Nope, it's just me being lazy and reusing what existed before.. I've fixed it now.